### PR TITLE
Added TextCNN wrapper TorchModel

### PR DESCRIPTION
**Pull request by Shiva-sankaran**
## Description

Following up on #3828. Second PR for #3828 , adds TextCNN wrapper TorchModel.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by rbharath — COMMENTED** [review]
Can you please review @shreyasvinaya

File: deepchem/models/torch_models/text_cnn.py
Comment (rbharath) [inline deepchem/models/torch_models/text_cnn.py:191]: Let's add this to the docs/

Comment (Shiva-sankaran) [reply deepchem/models/torch_models/text_cnn.py:191]: @shreyasvinaya I have a small doubt regarding updating docs. There is already an entry in the documentation for the TextCNNModel(https://deepchem.readthedocs.io/en/latest/api_reference/models.html#textcnnmodel), which currently links back to the TF implementation, should I replace it with this Torch version.

Comment (Shiva-sankaran) [reply deepchem/models/torch_models/text_cnn.py:191]: Also do I replace Keras with PyTorch in this table(https://deepchem.readthedocs.io/en/latest/api_reference/models.html#id134) as well? I don't see any entries of Keras/PyTorch

Comment (shreyasvinaya) [reply deepchem/models/torch_models/text_cnn.py:191]: add this to the docs under `torch_models`

Comment (Shiva-sankaran) [reply deepchem/models/torch_models/text_cnn.py:191]: Added to docs.

File: deepchem/models/torch_models/text_cnn.py
Comment (shreyasvinaya) [inline deepchem/models/torch_models/text_cnn.py:301]: Documentation on parameters to be added

File: deepchem/models/torch_models/text_cnn.py
Comment (shreyasvinaya) [inline deepchem/models/torch_models/text_cnn.py:320]: Documentation on parameters to be added

File: deepchem/models/torch_models/text_cnn.py
Comment (shreyasvinaya) [inline deepchem/models/torch_models/text_cnn.py:355]: Documentation on parameters to be added

File: deepchem/models/torch_models/text_cnn.py
Comment (shreyasvinaya) [inline deepchem/models/torch_models/text_cnn.py:380]: Documentation on parameters to be added
type annotations missing

File: deepchem/models/torch_models/text_cnn.py
Comment (shreyasvinaya) [inline deepchem/models/torch_models/text_cnn.py:387]: Documentation on parameters to be added
type annotations missing

**Comment by rbharath** [comment]
@Shiva-sankaran Before requesting a new review, make sure to respond to all remaining comments on the PR (can be thumbs up, or a sentence response even) so we know you have addressed all comments

File: deepchem/models/torch_models/text_cnn.py
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/text_cnn.py]: Added parameter descriptions.

File: deepchem/models/torch_models/text_cnn.py
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/text_cnn.py]: Added parameter descriptions.

File: deepchem/models/torch_models/text_cnn.py
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/text_cnn.py:389]: Added parameter descriptions.

File: deepchem/models/torch_models/text_cnn.py
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/text_cnn.py]: Added parameter descriptions and type annotations.

File: deepchem/models/torch_models/text_cnn.py
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/text_cnn.py]: Added parameter descriptions and type annotations.

**Comment by Shiva-sankaran** [comment]
> @Shiva-sankaran Before requesting a new review, make sure to respond to all remaining comments on the PR (can be thumbs up, or a sentence response even) so we know you have addressed all comments

Understood.

File: docs/source/api_reference/models.rst
Comment (rbharath) [inline docs/source/api_reference/models.rst:302]: Don't remove the old TF model; add this as a new model

**Review by rbharath — COMMENTED** [review]
One comment to fix on the docs and please rebase the PR

**Comment by rbharath** [comment]
@Shiva-sankaran Please don't merge; rebase instead. Merging messes up large PRs. Can you please rebase?

**Comment by Shiva-sankaran** [comment]
> @Shiva-sankaran Please don't merge; rebase instead. Merging messes up large PRs. Can you please rebase?

Yes, learnt it the hard way. Fixed it.

File: docs/source/api_reference/models.rst
Comment (Shiva-sankaran) [reply docs/source/api_reference/models.rst]: Reverted replacing the old TF model. Added the docs for the torch model under the Torch Models section.

**Review by rbharath — COMMENTED** [review]
@shreyasvinaya if this model looks good to you, can you LGTM it? I will go ahead and merge it in if it looks good to you

File: docs/source/api_reference/molecular_models.csv
Comment (shreyasvinaya) [inline docs/source/api_reference/molecular_models.csv:14]: Can you convert this to `Keras/PyTorch` as both implementations exist in the repo

**Comment by shreyasvinaya** [comment]
LGTM, ready to merge once the small docfix is done

File: docs/source/api_reference/molecular_models.csv
Comment (Shiva-sankaran) [reply docs/source/api_reference/molecular_models.csv]: Done.

**Review by shreyasvinaya — APPROVED** [review]
LGTM once CI passes
cc @rbharath

**Comment by rbharath** [comment]
@shreyasvinaya I'm seeing errors in docs-build python 3.11. Are these linked to this PR?

**Comment by shreyasvinaya** [comment]
@Shiva-sankaran can you please rebase the PR, i think the new commits should fix the doctest issues

**Comment by Shiva-sankaran** [comment]
@shreyasvinaya Rebased, there is still 2 doctest failures.

**Comment by shreyasvinaya** [comment]
@rbharath LGTM, good to merge

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**
